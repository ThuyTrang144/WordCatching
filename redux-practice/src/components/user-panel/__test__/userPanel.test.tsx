import { fireEvent, render } from "@testing-library/react";
import UserPanel from "..";

const USER_NAME = "Trang Nguyen";

describe("UserPanel component", () => {
  test("should render UserPanel component", () => {
    const { getByTestId } = render(<UserPanel username={USER_NAME} />);
    expect(getByTestId("user-panel")).toBeInTheDocument();
  });

  test("should not open menu when render UserPanel component", () => {
    const { queryByTestId } = render(<UserPanel username={USER_NAME} />);
    expect(queryByTestId("user-panel__menu")).not.toBeInTheDocument();
  });

  test("should open menu when click to avatar button", () => {
    const { getByTestId } = render(<UserPanel username={USER_NAME} />);
    const userPanelBtn = getByTestId("user-panel-btn");
    fireEvent.click(userPanelBtn);
    expect(getByTestId("user-panel__menu")).toBeInTheDocument();
  });

  test("should render nothing if username is undefined", () => {
    const { queryByTestId } = render(<UserPanel />);
    expect(queryByTestId("user-panel__loading-btn")).not.toBeInTheDocument();
  });

  test("matches snapshot when having username", () => {
    const { asFragment } = render(<UserPanel username={USER_NAME} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("matches snapshot when username is undefined", () => {
    const { asFragment } = render(<UserPanel />);
    expect(asFragment()).toMatchSnapshot();
  });
});
