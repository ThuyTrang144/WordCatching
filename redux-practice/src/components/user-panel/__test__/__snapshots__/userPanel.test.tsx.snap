// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPanel component matches snapshot when having username 1`] = `
<DocumentFragment>
  <div
    class="user-panel"
  >
    <div
      class="user-panel__avt-group"
      data-testid="user-panel"
    >
      <button
        class="chakra-button user-panel__btn css-4xx2wk"
        data-testid="user-panel-btn"
        type="button"
      >
        <div
          class="user-panel__avatar"
        >
          <span>
            TN
          </span>
        </div>
        <span
          class="user-panel__username"
          data-testid="username"
        >
          Trang Nguyen
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`UserPanel component matches snapshot when username is undefined 1`] = `
<DocumentFragment>
  <div
    class="user-panel"
  >
    <div
      class="user-panel__avt-group"
      data-testid="user-panel"
    />
  </div>
</DocumentFragment>
`;
